FreeRTOS-Plus/Demo/FreeRTOS-IoT-Libraries-LTS-Beta2/mqtt/mqtt_multitask/DemoTasks/MultitaskMQTTExample.c

```diff
@@ -0,0 +1,1457 @@
+                    {
+                        LogInfo( ( "Adding subscription to %.*s",
+                                   pxSubscribeInfo[ i ].topicFilterLength,
+                                   pxSubscribeInfo[ i ].pTopicFilter ) );
+                        LogInfo( ( "Filter length: %u", pxSubscribeInfo[ i ].topicFilterLength ) );
+                        prvAddSubscription( pxSubscribeInfo[ i ].pTopicFilter,
+                                            pxSubscribeInfo[ i ].topicFilterLength,
+                                            xAckInfo.pCommandContext->pResponseQueue );
+                    }
+                }
+                else
+                {
+                    xStatus = MQTTBadResponse;
+                }
+
+                xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                if( xAckInfo.callback != NULL )
+                {
+                    xAckInfo.callback( xAckInfo.pCommandContext );
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_UNSUBACK:
+                xAckInfo = prvPopAck( packetIdentifier );
+
+                if( xAckInfo.packetId == packetIdentifier )
+                {
+                    pxSubscribeInfo = xAckInfo.pCommandContext->pSubscribeInfo;
+
+                    for( i = 0; i < xAckInfo.pCommandContext->subscriptionCount; i++ )
+                    {
+                        LogInfo( ( "Removing subscription to %.*s",
+                                   pxSubscribeInfo[ i ].topicFilterLength,
+                                   pxSubscribeInfo[ i ].pTopicFilter ) );
+                        prvRemoveSubscription( pxSubscribeInfo[ i ].pTopicFilter,
+                                               pxSubscribeInfo[ i ].topicFilterLength,
+                                               xAckInfo.pCommandContext->pResponseQueue );
+                    }
+                }
+                else
+                {
+                    xStatus = MQTTBadResponse;
+                }
+
+                xAckInfo.pCommandContext->returnStatus = xStatus;
+
+                if( xAckInfo.callback != NULL )
+                {
+                    xAckInfo.callback( xAckInfo.pCommandContext );
+                }
+
+                break;
+
+            case MQTT_PACKET_TYPE_PUBREC:
+            case MQTT_PACKET_TYPE_PUBREL:
+                break;
+
+            case MQTT_PACKET_TYPE_PINGRESP:
+
+                /* Nothing to be done from application as library handles
+                 * PINGRESP. */
+                LogWarn( ( "PINGRESP should not be handled by the application "
+                           "callback when using MQTT_ProcessLoop.\n\n" ) );
+                break;
+
+            /* Any other packet type is invalid. */
+            default:
+                LogError( ( "Unknown packet type received:(%02x).\n\n",
+                            pPacketInfo->type ) );
```

Comment:
What to be done in this case? I think we would need to assert in here in order to enable debugging from this point. Ideally connection should be ignored and need to be reconnected I think. Also, we would need to update other demos as well for this scenario.